# number of particles
M: 100

# platform volume (i.e., states/particles outside this volume are invalid)
platform_volume:
  x: [-0.150, 0.150] # platform width
  y: [-0.175, 0.175] # platform length
  z: [ 0.000, 0.390] # rod length (height)
  R: [-0.875, 0.875] # -50 to 50 degrees
  P: [-0.875, 0.875] # -50 to 50 degrees
  Y: [-3.142, 3.142]

# When the particles are first initialized, they will be uniformly distributed
# within the `initial_volume` bounding box. Specifically, the z-component of
# particles do not need a large range since we assume that MCL starts when the
# quadrotor is not flying and is level on the base of the platform.
initial_volume:
  x: [-0.050, 0.050]
  y: [-0.050, 0.050]
  z: [ 0.370, 0.390]
  # R: [-0.08726, 0.08726] # -5 to 5 degrees
  # P: [-0.08726, 0.08726] # -5 to 5 degrees
  R: [3.142, 3.142] # -eps to eps degrees
  P: [0.00, 0.00] # -eps to eps degrees
  Y: [-1.56, -1.58]
  # Y: [-3.142, 3.142]

# motion model
#   0 - stochastically-driven nearly-constant velocity
#   1 - mechanized motion model w/ IMU
#
#   (defn come from `MotionModels` enum in motion_model.h)
motion_model: 1

# parameters for the SDNCV motion model
sdncv:
  # standard deviation parameters for linear and angular acceleration
  x:  15
  y:  15
  z:  15
  ax: 200
  ay: 200
  az: 200

# parameters for the IMU mechanized motion model
mech:
  # standard deviation parameters for linear and angular velocity
  x:  0.0075
  y:  0.0075
  z:  0.0075
  ax: 0.01
  ay: 0.01
  az: 0.01
